features/FEATURE_BLE/ble/pal/MemorySecurityDB.h

```diff
@@ -0,0 +1,373 @@
+        if (entry) {
+            entry->state = ENTRY_WRITTEN;
+            entry->local_keys.ediv = ediv;
+            entry->local_keys.rand = rand;
+        }
+    }
+
+    /* peer's keys */
+
+    /* get */
+    virtual void get_entry_peer_csrk(
+        SecurityEntryCsrkDbCb_t cb,
+        entry_handle_t entry_handle
+    ) {
+        csrk_t csrk;
+        entry_t *entry = as_entry(entry_handle);
+        if (entry) {
+            csrk = entry->csrk;
+        }
+        cb(entry_handle, &csrk);
+    }
+
+    virtual void get_entry_peer_keys(
+        SecurityEntryKeysDbCb_t cb,
+        entry_handle_t entry_handle
+    ) {
+        SecurityEntryKeys_t *key = NULL;
+        entry_t *entry = as_entry(entry_handle);
+        if (entry) {
+            key = &entry->peer_keys;
+        }
+        cb(entry_handle, key);
+    }
+
+    /* set */
+
+    virtual void set_entry_peer_ltk(
+        entry_handle_t entry_handle,
+        const ltk_t &ltk
+    ) {
+        entry_t *entry = as_entry(entry_handle);
+        if (entry) {
+            entry->state = ENTRY_WRITTEN;
+            entry->peer_keys.ltk = ltk;
+        }
+    }
+
+    virtual void set_entry_peer_ediv_rand(
+        entry_handle_t entry_handle,
+        const ediv_t &ediv,
+        const rand_t &rand
+    ) {
+        entry_t *entry = as_entry(entry_handle);
+        if (entry) {
+            entry->state = ENTRY_WRITTEN;
+            entry->peer_keys.ediv = ediv;
+            entry->peer_keys.rand = rand;
+        }
+    }
+
+    virtual void set_entry_peer_irk(
+        entry_handle_t entry_handle,
+        const irk_t &irk
+    ) {
+        entry_t *entry = as_entry(entry_handle);
+        if (entry) {
+            entry->state = ENTRY_WRITTEN;
+            entry->peer_identity.irk = irk;
+        }
+    }
+
+    virtual void set_entry_peer_bdaddr(
+        entry_handle_t entry_handle,
+        bool address_is_public,
+        const address_t &peer_address
+    ) {
+        entry_t *entry = as_entry(entry_handle);
+        if (entry) {
+            entry->state = ENTRY_WRITTEN;
+            entry->peer_identity.identity_address = peer_address;
+        }
+    }
+
+    virtual void set_entry_peer_csrk(
+        entry_handle_t entry_handle,
+        const csrk_t &csrk
+    ) {
+        entry_t *entry = as_entry(entry_handle);
+        if (entry) {
+            entry->state = ENTRY_WRITTEN;
+            entry->csrk = csrk;
+        }
+    }
+
+    /* local csrk */
+
+    virtual const csrk_t* get_local_csrk() {
+        return &_local_csrk;
+    }
+
+    virtual void set_local_csrk(const csrk_t &csrk) {
+        _local_csrk = csrk;
+    }
+
+    /* public key */
+
+    virtual const public_key_t& get_public_key_x() {
+        return _public_key_x;
+    }
+
+    virtual const public_key_t& get_public_key_y() {
+        return _public_key_y;
+    }
+
+    virtual void set_public_key(
+        const public_key_t &public_key_x,
+        const public_key_t &public_key_y
+    ) {
```

Comment:
`public_key_x` and `public_key_y` are the coordinates of the point that forms the public key. `public_key_t` should reflect that - either is should contain both coordinates or be renamed to something like `public_key_coordinate_t`.

Reply:
renamed